docs/DataTypes.md

```diff
@@ -0,0 +1,40 @@
+# Data Types
+
+DataHelix currently recognises three distinct data types. Keeping this set small is a deliberate goal; it would be possible to have types like _FirstName_ or _Postcode_, but instead these are considered specialisations of the _String_ type, so they can be constrained by the normal string operators (e.g. a user could generate all first names shorter than 10 characters, starting with a vowel).
+
+## Numeric
+
+In principle, any real number. In practice, any number that can be represented in a Java [BigDecimal](https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html) object.
+
+In profile files, numbers must be expressed as JSON numbers, without quotation marks.
+
+### Granularity
+
+The granularity of a numeric field is a measure of how small the distinctions in that field can be; it is the smallest positive number of which every valid value is a multiple. For instance:
+
+- if a numeric field has a granularity of 1, it can only be satisfied with multiples of 1, a.k.a. integers
+- if a numeric field has a granularity of 0.1, it can be satified by 1, or 1.1, but not 1.11
+
+At present, only granularities less than or equal to 1 are supported (in future, it may be possible to specify higher granularities - a granularity of 2 would admit only even numbers). Additionally, granularities must be powers of ten, so 0.01 is legal but 0.02 is not.
```

Comment:
in a different comment, predictive documentation was disagreed with. her we say that  `"(in future, it may be possible to specify higher granularities"`.
I would agree with omitting possible future functionality.

Reply:
Hmm, good point. I think the example of granularity=2 does aid understanding, but you're right that it's a bit crystal-ball-esque as it stands. How about:

> Granularities must be powers of ten less than or equal to one (1, 0.1, 0.01, etc). Granularities outside these restrictions could be potentially useful (e.g. a granularity of 2 would permit only even numbers) but are not currently supported.

Reply:
thats sounds fine.